lib/puppet/configurer.rb

```diff
@@ -328,6 +353,31 @@ def run_internal(options)
   end
   private :run_internal
 
+  def find_functional_server()
+    configured_environment = Puppet[:environment] if Puppet.settings.set_by_config?(:environment)
+
+    Puppet.settings[:server_list].find do |server|
+      # Puppet.override doesn't return the result of its block, so we
+      # need to handle this manually
+      found = false
+      Puppet.override(:server => server[0], :serverport => server[1]) do
+        begin
+          Puppet::Node.indirection.find(Puppet[:node_name_value],
```

Comment:
This appears to introduce an additional node request. Is there any way we can return that and avoid calling it later?